### PR TITLE
Scenario/scn3 diff az

### LINKED ISSUE #229: Change scenario3 infrastructure
We need to show disk statistics for different availability zones

### PR DESCRIPTION
**Pull request by anton-sidelnikov**
Changes according request after presentation
Close #229

### CODE REVIEW

File: scenarios/scenario3/variables.tf [hunk 1]
```diff
@@ -1,7 +1,7 @@
-variable "region" { default = "eu-de"}
+variable "region" { default = "eu-de" }
 variable "availability_zone" { default = "eu-de-03" }
 variable "ecs_flavor" {}
 variable "ecs_image" {}
 variable "addr_3_octets" { default = "192.168.0" }
 variable "scenario" {}
-variable "public_key" {}
\ No newline at end of file
+variable "public_key" { }
```
Comment (lego963) [inline scenarios/scenario3/variables.tf:7]: trailing whitespaces here and region

File: scenarios/scenario3/nodes/variables.tf [hunk 2]
```diff
@@ -0,0 +1,25 @@
+variable "net_address" {}
+variable "ecs_flavor" {}
+variable "ecs_image" {}
+variable "bastion_sec_group_id" {}
+variable "network_id" {}
+variable "subnet_id" {}
+variable "scenario" {}
+variable "key_pair_name" {}
+variable "availability_zone" {
```
Comment (outcatcher) [inline scenarios/scenario3/nodes/variables.tf:9]: Let's rename it to "availability_zones"

File: scenarios/scenario3/nodes/variables.tf [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    type    = list(string)
+    default = ["eu-de-01", "eu-de-02","eu-de-03"]
+}
+variable "nodes_count" { default = 3 }
+variable "disc_volume" {
+  type    = number
+  default = 5
+}
+variable "volume_type" {
```
Comment (outcatcher) [inline scenarios/scenario3/nodes/variables.tf:18]: `volume_types`?

File: scenarios/scenario3/nodes/vm.tf [hunk 4]
```diff
@@ -0,0 +1,56 @@
+data "opentelekomcloud_images_image_v2" "current_image" {
+  name        = var.ecs_image
+  most_recent = true
+}
+
+# Create instance
+resource "opentelekomcloud_compute_instance_v2" "http" {
+  count       = length(opentelekomcloud_networking_port_v2.http)
+  name        = "${local.workspace_prefix}${var.scenario}_eu-de-0${count.index + 1}"
```
Comment (outcatcher) [inline scenarios/scenario3/nodes/vm.tf:9]: Maybe use `var.availability_zone[count.index]` also in row 9?

File: scenarios/scenario3/nodes/volumes.tf [hunk 5]
```diff
@@ -0,0 +1,90 @@
+# Create volumes
+resource "opentelekomcloud_blockstorage_volume_v2" "volumes_az1" {
+  count             = 3
+  name              = format("vol-%s", "${var.volume_type[count.index]}_${opentelekomcloud_compute_instance_v2.http[0].name}")
```
Comment (outcatcher) [inline scenarios/scenario3/nodes/volumes.tf:4]: Why you need a format here?

File: scenarios/scenario3/nodes/volumes.tf [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+  instance_id = opentelekomcloud_compute_instance_v2.http[0].id
+  volume_id   = opentelekomcloud_blockstorage_volume_v2.volumes_az1[2].id
+  depends_on = [
+    opentelekomcloud_compute_volume_attach_v2.attach-az1-1
+  ]
+}
+
+resource "opentelekomcloud_blockstorage_volume_v2" "volumes_az2" {
+  count             = 3
+  name              = format("vol-%s", "${var.volume_type[count.index]}_${opentelekomcloud_compute_instance_v2.http[1].name}")
```
Comment (outcatcher) [inline scenarios/scenario3/nodes/volumes.tf:34]: And here

File: scenarios/scenario3/nodes/volumes.tf [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+  ]
+}
+
+resource "opentelekomcloud_blockstorage_volume_v2" "volumes_az3" {
+  count             = 3
+  name              = format("vol-%s", "${var.volume_type[count.index]}_${opentelekomcloud_compute_instance_v2.http[2].name}")
+  size              = 2
+  volume_type       = var.volume_type[count.index]
+  availability_zone = var.availability_zone[2]
+}
```
Comment (outcatcher) [inline scenarios/scenario3/nodes/volumes.tf:68]: Can we make it in a cycle?

Comment (anton-sidelnikov) [reply scenarios/scenario3/nodes/volumes.tf:68]: No, in cicle i can't control disk assigments

**Pull request merged**